chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -251,6 +259,7 @@ case class ChainHandler(
       val maxHeight = filterHeadersToCreate.maxBy(_.height)
       logger.info(
         s"Processed filters headers from height=${minHeight.height} to ${maxHeight.height}. Best hash=${maxHeight.blockHashBE}")
+      cacheFilterHeaderCount(None)
```

Comment:
You need to do this for the block count and filter count as well

Reply:
Every time `ChainHandler.processHeaders` processes a new block header it creates a new instance of `ChainHandler`: https://github.com/bitcoin-s/bitcoin-s/blob/master/chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala#L113. So there's no need in resetting the block count cache.

Reply:
Should update the filter count cache still

Reply:
> Every time `ChainHandler.processHeaders` processes a new block header it creates a new instance of `ChainHandler`: https://github.com/bitcoin-s/bitcoin-s/blob/master/chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala#L113. So there's no need in resetting the block count cache.

This is an important design goal I had in mind, if we can "encapsulate" state changes by returning new interface types (in this case `Future[ChainApi]`) we can get away with internal mutability. However in certain parts of our codebase we discard the returned `ChainApi` ([here with wallet/node](https://github.com/bitcoin-s/bitcoin-s/blob/930f184dca69a3aa7512fc92a3db6833c5bfac6d/app/server/src/main/scala/org/bitcoins/server/Main.scala#L119-L120)). I really like the pattern of returning new `Api` types to signify a "state change" of the internal API, but I guess I haven't been explicit about it as a design goal.